docs/use-cases-orchestration.md

```diff
@@ -4,15 +4,57 @@ title: Microservice Orchestration and Saga
+The implementation must guarantee that all of the calls must eventually succeed even with the occurrence of prolonged failures of individual microservices.
 
- * [Using Temporal workflows to spin up Kubernetes by Banzai Cloud](https://banzaicloud.com/blog/introduction-to-cadence/)
- * [Improving the User Experience with Uber’s Customer Obsession Ticket Routing Workflow and Orchestration Engine](https://eng.uber.com/customer-obsession-ticket-routing-workflow-and-orchestration-engine/)
+In some cases, instead of trying to complete the process by retrying for a long time, compensation rollback logic should be executed. [Saga Pattern](https://microservices.io/patterns/data/saga.html) is one way to standardize compensation APIs.
+
+## Benefits
+
+Temporal is a perfect fit for such scenarios. It guarantees that workflow code eventually completes, has built-in support for unlimited exponential activity retries, and simplifies the coding of the compensation logic. 
+
+Temporal also gives full visibility into the state of each workflow, in contrast to an ad-hoc orchestration based on queues where getting a current status of each request is virtually impossible.
+
+## Example
+
+The workflow snippet below orchestrates two activity calls: booking a hotel and reserving a flight. If the first activity fails (including all the configured retries), the workflow returns directly.
+
+However, if the first reservation succeeds, but the second one fails, you may want to cancel the hotel to avoid undesired charges. That's why the error-handling block contains a third activity call that cancels the hotel reservation before completing the workflow.
+
+```go
+// Step 1: Book a hotel.
+var hotel HotelResult
+err := workflow.ExecuteActivity(ctx, activities.BookHotel, tripRequest).Get(ctx, &hotel)
+if err != nil {
+    // If the hotel reservation failed, fail the workflow.
+    return err
+}
+
+
+// Step 2: Book a flight.
+var flight FlightResult
+err := workflow.ExecuteActivity(ctx, activities.BookFlight, tripRequest).Get(ctx, &flight)
+if err != nil {
+    // If the flight reservation failed, cancel the hotel.
+    var cancelResult HotelCancellationResult
+    cancelErr := workflow.ExecuteActivity(ctx, activities.CancelHotel, hotel).Get(ctx, &cancelResult)
+    if cancelErr != nil {
+        // The hotel cancellation failed... probably, some manual action is needed.
+        return cancelErr
+    }
+    
+    return err
+}
+
+// Both reservations succeeded.
+```
+
+## Next Steps
+
+Here are two real-world examples of Temporal-based service orchestration scenarios:
+
+ * [Using Temporal workflows to spin up Kubernetes](https://banzaicloud.com/blog/introduction-to-cadence/) by Banzai Cloud
+ * Improving the User Experience with Uber’s [Customer Obsession Ticket Routing Workflow and Orchestration Engine](https://eng.uber.com/customer-obsession-ticket-routing-workflow-and-orchestration-engine/)
```

Comment:
Add link to Java Saga Sample: https://github.com/temporalio/temporal-java-samples/tree/master/src/main/java/io/temporal/samples/bookingsaga